airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt

```diff
@@ -0,0 +1,450 @@
+import io.airbyte.commons.json.Jsons.emptyObject
+import io.airbyte.integrations.base.destination.typing_deduping.AirbyteProtocolType
+import io.airbyte.integrations.base.destination.typing_deduping.AirbyteType
+import io.airbyte.integrations.base.destination.typing_deduping.Array
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationInitialStatus
+import io.airbyte.integrations.base.destination.typing_deduping.InitialRawTableStatus
+import io.airbyte.integrations.base.destination.typing_deduping.Sql
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId
+import io.airbyte.integrations.base.destination.typing_deduping.Struct
+import io.airbyte.integrations.base.destination.typing_deduping.Union
+import io.airbyte.integrations.base.destination.typing_deduping.UnsupportedOneOf
+import io.airbyte.integrations.destination.snowflake.SnowflakeDatabaseUtils.toSqlTypeName
+import io.airbyte.integrations.destination.snowflake.migrations.SnowflakeState
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair
+import io.airbyte.protocol.models.v0.DestinationSyncMode
+import java.sql.Connection
+import java.sql.DatabaseMetaData
+import java.sql.ResultSet
+import java.sql.SQLException
+import java.time.Instant
+import java.util.*
+import net.snowflake.client.jdbc.SnowflakeSQLException
+import org.apache.commons.text.StringSubstitutor
+import org.jooq.SQLDialect
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class SnowflakeDestinationHandler(
+    databaseName: String,
+    private val database: JdbcDatabase,
+    rawTableSchema: String
+) :
+    JdbcDestinationHandler<SnowflakeState>(
+        databaseName,
+        database,
+        rawTableSchema,
+        SQLDialect.POSTGRES
+    ) {
+    // Postgres is close enough to Snowflake SQL for our purposes.
+    // We don't quote the database name in any queries, so just upcase it.
+    private val databaseName = databaseName.uppercase(Locale.getDefault())
+
+    @Throws(SQLException::class)
+    private fun getFinalTableRowCount(
+        streamIds: List<StreamId>
+    ): LinkedHashMap<String, LinkedHashMap<String, Int>> {
+        val tableRowCounts = LinkedHashMap<String, LinkedHashMap<String, Int>>()
+        // convert list stream to array
+        val namespaces = streamIds.map { it.finalNamespace }.toTypedArray()
+        val names = streamIds.map { it.finalName }.toTypedArray()
+        val query =
+            """
+                         SELECT table_schema, table_name, row_count
+                         FROM information_schema.tables
+                         WHERE table_catalog = ?
+                           AND table_schema IN (${IntRange(1, streamIds.size).joinToString { "?" }})
+                           AND table_name IN (${IntRange(1, streamIds.size).joinToString { "?" }})
+                         
+                         """.trimIndent()
```

Comment:
Line 62 - 72 is rewrite of this 
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java#L105-L115